devenv/main.tf

```diff
@@ -165,41 +173,101 @@ resource "azurerm_role_assignment" "approutingdnszone" {
   principal_id         = data.azurerm_user_assigned_identity.clusteridentity.principal_id
 }
 
-resource "local_sensitive_file" "kubeconfig" {
-  content  = azurerm_kubernetes_cluster.cluster.kube_config_raw
-  filename = "${path.module}/state/kubeconfig"
+resource "azurerm_container_registry" "acr" {
+  name                = "approutingdev${random_string.random.result}a"
```

Comment:
Is the "a" there just in case the random string ends in an invalid character?

Reply:
Yes, resource names (at least for some RPs?) can't end in a number